src/scripts/push_coral_to_hub.py

```diff
@@ -141,22 +206,101 @@ def main(
     )
 
-    # Add the speaker metadata to the recordings
+    # Add the speaker metadata to the recordings. Since there possibly are two speakers
+    # in each recording, we add the metadata for both speakers to the recordings. We
+    # rename the columns to avoid conflicts.
+    test_recordings_df = test_recordings_df.merge(
+        speaker_metadata.rename(columns=lambda x: f"{x}_1"),
+        on="speaker_id_1",
+        how="left",
+    )
     test_recordings_df = test_recordings_df.merge(
-        speaker_metadata, left_on="speaker_id", right_on="speaker_id"
+        speaker_metadata.rename(columns=lambda x: f"{x}_2"),
+        on="speaker_id_2",
+        how="left",
+    )
+    train_recordings_df = train_recordings_df.merge(
+        speaker_metadata.rename(columns=lambda x: f"{x}_1"),
+        how="left",
     )
     train_recordings_df = train_recordings_df.merge(
-        speaker_metadata, left_on="speaker_id", right_on="speaker_id"
+        speaker_metadata.rename(columns=lambda x: f"{x}_2"),
+        how="left",
     )
 
+    # Pick a validation set from the training set, by selecting a single recording
+    # from each speaker. We do this by selecting the last recording from each speaker
+    # in the training set, as the first recordings are conversations.
+    validation_recordings = []
+    for speaker_id in train_recordings_df["speaker_id_1"].unique():
+        validation_recordings.append(
+            train_recordings_df[train_recordings_df["speaker_id_1"] == speaker_id].iloc[
+                -1
+            ]
+        )
+    validation_recordings_df = pd.DataFrame(validation_recordings).astype(str)
+
+    # Remove the validation recordings from the training set
+    train_recordings_df = train_recordings_df[
+        ~train_recordings_df["filename"].isin(validation_recordings_df["filename"])
+    ].astype(str)
+
     # Create the dataset
     testset = Dataset.from_pandas(test_recordings_df).cast_column("filename", Audio())
     trainset = Dataset.from_dict(train_recordings_df).cast_column("filename", Audio())
-    dataset_dict = DatasetDict({"train": trainset, "test": testset})
+    validationset = Dataset.from_dict(validation_recordings_df).cast_column(
+        "filename",
+        Audio(),
+    )
+    dataset_dict = DatasetDict(
+        {"train": trainset, "test": testset, "validation": validationset}
+    )
 
     # Create hub-id
     hub_id = f"{hub_id}_v{major_version}.{minor_version}"
 
+    # Check if the dataset already exists on the hub, and if so, prompt the user to
+    # update the version number
+    api = HfApi()
+    try:
+        dataset_info = api.dataset_info(repo_id=hub_id)
+        if dataset_info.id == hub_id:
+            logger.error(
+                (
+                    f"The dataset {hub_id} already exists on the hub!",
+                    "Please update the version number.",
+                )
+            )
+            return
```

Comment:
Maybe just raise an error here directly? Same thing with the next `logger.error`.